### PR TITLE
v1.1.0 doc changes

### PR DESCRIPTION
**Pull request by van-fs**
Updated docs to reflect V1.1.0.  @TrevorFSmith can you check the fan-out doc?  I'm trying it in an integration test, and this is the debug output:

```
dataLayer[0].ecommerce.promoView.promotions handleData entry
[{"id":"1004-Blueberries123321","name":"Fruits","creative":"Blueberries123321","position":"Feature"}] 1.1.0.js:1:18535
dataLayer[0].ecommerce.promoView.promotions handleData exit
[] 1.1.0.js:1:18535
dataLayer[0].ecommerce.promoView.promotions handleData entry
[{"id":"1001-Strawberries222333","name":"Fruits","creative":"Strawberries222333","position":"Sub1"}] 1.1.0.js:1:18535
dataLayer[0].ecommerce.promoView.promotions handleData exit
[]
```

It doesn't seem like it executes the handler chain.  Here's the data layer 

```html
  <script>
    dataLayer = [
      {
        event: 'impressions_loaded',
        ecommerce: {
          promoView: {
            promotions: [
              {
                id: '1004-Blueberries123321',
                name: 'Fruits',
                creative: 'Blueberries123321',
                position: 'Feature',
              },
              {
                id: '1001-Strawberries222333',
                name: 'Fruits',
                creative: 'Strawberries222333',
                position: 'Sub1',
              },
            ]
          }
        },
        'gtm.uniqueEventId': 6
      },
    ];
  </script>
```

with the rule

```javascript
      {
        source: 'dataLayer[0].ecommerce.promoView.promotions',
        operators: [{ name: 'fan-out' }, { name: 'insert', value: 'Viewed Promotion' }],
        destination: 'FS.event',
        debug: true
      }
```

I'm doing something wrong I suppose.

### CODE REVIEW

**Comment by TrevorFSmith** [comment]
> ```js
>       {
>         source: 'dataLayer[0].ecommerce.promoView.promotions',
>         operators: [{ name: 'fan-out' }, { name: 'insert', value: 'Viewed Promotion' }],
>         destination: 'FS.event',
>         debug: true
>       }
> ```

When the source is an array with no index specified then you don't need the fan-out operator, it's implied.

**Comment by van-fs** [comment]
So it does, NICE!  Updated the docs to demonstrate the implicit and explicit uses cases.  This should cap the v1.1.0 docs.

**Review by TrevorFSmith — CHANGES_REQUESTED** [review]
I think we need to change the fan-out doc to mention both the automagical support for array sources (without the fan-out) and that fan-out is usually necessary if a previous operator did some data prep or filtering.

File: docs/operator_fan-out.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+   position: 'Sub1'
+   }
+]
+```
+
+## Sending multiple events from an object's properties
+
+```javascript
+{
+  source: 'dataLayer[0].ecommerce.promoView.promotions',
```
[the remark this answers is not in the document]
Comment (van-fs) [reply docs/operator_fan-out.md:85]: I made a (perhaps too subtle) change to the input.  `promotions` is an object now with numeric (coerced as string) properties.  So it's `promotions.0` and `promotions.1` that are fanned out.

**Review by TrevorFSmith — APPROVED** [review]

**Pull request merged**